include/enumerations/material_definitions.hpp

```diff
@@ -0,0 +1,199 @@
+#pragma once
```

Comment:
I think I commented this on the boundary PR, are you gonna add some documentation here?

Reply:
Yes, put some documentation updates in the other branch.